apis/python/src/tiledbsoma/_multiscale_image.py

```diff
@@ -0,0 +1,257 @@
+#
+# Copyright (c) 2024 The Chan Zuckerberg Initiative Foundation
+# Copyright (c) 2024 TileDB, Inc.
+#
+# Licensed under the MIT License.
+"""
+Implementation of a SOMA MultiscaleImage.
+"""
+
+import warnings
+from typing import Any, Optional, Sequence, Tuple, Union
+
+import pyarrow as pa
+import somacore
+from somacore import CoordinateSpace, CoordinateTransform, ScaleTransform, options
+from typing_extensions import Self
+
+from ._constants import SPATIAL_DISCLAIMER
+from ._dense_nd_array import DenseNDArray
+from ._soma_object import AnySOMAObject
+from .options import SOMATileDBContext
+
+
+class ImageProperties:
+    """Properties for a single resolution level in a multiscale image.
+
+    Lifecycle:
+        Experimental.
+    """
+
+    @property
+    def name(self) -> str:
+        """The key for the image.
+
+        Lifecycle:
+            Experimental
+        """
+        raise NotImplementedError()
+
+    @property
+    def shape(self) -> Tuple[int, ...]:
+        """Size of each axis of the image.
+
+        Lifecycle:
+            Experimental.
+        """
+        raise NotImplementedError()
+
+
+class MultiscaleImage(somacore.MultiscaleImage[DenseNDArray, AnySOMAObject]):
+    """A multiscale image with an extendable number of resolution levels.
+
+    The multiscale image defines the top level properties. Each level must
+    match the expected following properties:
+    * number of channels
+    * axis order
```

Comment:
```suggestion
    """A multiscale image represented as a collection of images at multiple resolution levels.

    Each level of the multiscale image must have the following consistent properties:

    * **Number of Channels**: All levels must have the same number of channels.
    * **Axis Order**: The order of axes (e.g., channels, height, width) must be consistent across levels.
```

I wanted to highlight that the tiledb implementation is a _collection_ of images and attempted to be a little clearer about the fact that the listed properties must have have identical values.